File: 준영/LV2/예상 대진표.js
function solution(n,a,b) {
  var answer = 1
  var a_seed = a % 2 ? parseInt(a / 2) + 1 : a / 2
  var b_seed = b % 2 ? parseInt(b / 2) + 1 : b / 2

  while (n != 1){
    if (a_seed === b_seed) break

    a_seed = a_seed % 2 ? parseInt(a_seed / 2) + 1 : a_seed / 2
    b_seed = b_seed % 2 ? parseInt(b_seed / 2) + 1 : b_seed / 2
    answer += 1
    n /= 2
  }
  return answer
}

File: 준영/프로그래머스/LV2/예상 대진표.py
def solution(n,a,b):
    answer = 1
    # a_seed = (a // 2) + 1 if a % 2 else a // 2
    a_seed = (a // 2) + (a % 2)
    b_seed = (b // 2) + 1 if b % 2 else b // 2
    
    while n != 1:
        if a_seed == b_seed:
            break

        a_seed = (a_seed // 2) + 1 if a_seed % 2 else a_seed // 2
        b_seed = (b_seed // 2) + 1 if b_seed % 2 else b_seed // 2
        
        answer += 1
        n //= 2
    return answer

solution(8,4,7)
print(solution(8, 4, 7))